### PR TITLE
Stat Calculator

### PR DESCRIPTION
**Pull request by slabserif**
Started laying ground work for how a pokemon's stats are initially
created, and how they are modified upon levelling up.

### CODE REVIEW

File: src/Core/PokemanzUtil.cs [hunk 1]
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
...
+        //QUESTIONS:
+        //--BaseStats found online seem too high and lack clarity. Are these stats assuming the pokemon is at level 1, 0, 100, or what? 
+        //--Is the stat calculation meant to add on to an existing stat, or replace it?
+        //--How do Exp points fit into this and what determines how much exp is needed per level?
+
+        //Dv randomizer is used in generating the pokemon's initial stats, as well as making levelling up produce diverse stats between two of the same pokemon at the same level.
+        //Dv reserach is derived from: http://bulbapedia.bulbagarden.net/wiki/Individual_values
+        /*static int DvRandomizer(int min, int max)
+        {
+            Random random = new Random();
```
Comment (Gekctek) [inline src/Core/PokemanzUtil.cs]: we will make this a global instead of recreating it every time

File: src/Core/PokemanzUtil.cs [hunk 2]
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
...
+            int dv = random.Next(0, 15);
```
Comment (Gekctek) [inline src/Core/PokemanzUtil.cs]: 15?

File: src/Core/PokemanzUtil.cs [hunk 3]
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
...
+        //--New moves if any (NOT MVP)
+        //--Evolution (NOT MVP)
+        //If hits level 100, cannot level anymore (NOT MVP)
+
+        //QUESTIONS:
+        //--BaseStats found online seem too high and lack clarity. Are these stats assuming the pokemon is at level 1, 0, 100, or what? 
+        //--Is the stat calculation meant to add on to an existing stat, or replace it?
+        //--How do Exp points fit into this and what determines how much exp is needed per level?
+
+        //Dv randomizer is used in generating the pokemon's initial stats, as well as making levelling up produce diverse stats between two of the same pokemon at the same level.
```
Comment (Gekctek) [inline src/Core/PokemanzUtil.cs]: dv?

File: src/Core/PokemanzUtil.cs [hunk 4]
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
...
+        //Dv reserach is derived from: http://bulbapedia.bulbagarden.net/wiki/Individual_values
+        /*static int DvRandomizer(int min, int max)
+        {
+            Random random = new Random();
+            int dv = random.Next(0, 15);
+            return dv;
+        }
+
+        public int CalcNewStat(int pokemon.basestat, int dv)
```
Comment (Gekctek) [inline src/Core/PokemanzUtil.cs]: just call it base stat

File: src/Core/PokemanzUtil.cs [hunk 5]
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
...
+        {
+            decimal newStat = ((((basestat + dv) * 2) + ((Math.Sqrt(Ev) / 4) * level) / 100) + 5;
+            //TODO: round down decimal newHp, and convert to int
```
Comment (Gekctek) [inline src/Core/PokemanzUtil.cs]: Math.Round will do that

File: src/Core/PokemanzUtil.cs [hunk 6]
```diff
@@ -8,6 +8,49 @@ namespace Pokemanz.Core
...
+            return int newStat;
+        }
+
+        public int CalcNewHp(int pokemon.hp, int dv)
+        {
+            decimal newHp = ((((basestat + dv) * 2) + ((Math.Sqrt(Ev) / 4) * level) / 100) + level + 10;
```
Comment (Gekctek) [inline src/Core/PokemanzUtil.cs]: looks like a copy and paste job, combine them

File: src/Core/Pokemon.cs [hunk 7]
```diff
@@ -21,6 +21,7 @@ public class Pokemon
         public int SpAttack { get; set; }
         public int SpDefense { get; set; }
         public int Speed { get; set; }
+        public int Ev { get; set; }
```
Comment (Gekctek) [inline src/Core/Pokemon.cs]: ev?

File: src/Core/compiler/resources/pokedex.txt [hunk 8]
```diff
@@ -1,11 +1,11 @@
 Id	Name	Type 1	Type 2	Level	Hp	Attack	Defense	SpAttack	SpDefense	Speed
-1	Bulbasuar	Grass		5	21	12	11	11	13	11
+1	Bulbasuar	Grass			45	49	49	65	65	45
```
Comment (Gekctek) [inline src/Core/compiler/resources/pokedex.txt]: why are the stats so much bigger for this one

File: src/Core/PokemanzUtil.cs (hunk 2 above)
Comment (slabserif) [inline src/Core/PokemanzUtil.cs]: Should be changed to 16 since it's an exclusive maximum

**Comment by Gekctek** [comment]
I will pull this once it is fixed and when something is acutally done. as long as it is in a branch and pushed/sync'd with the server it is good. only pull request when it is done

File: src/Core/compiler/resources/pokedex.txt (hunk 8 above)
Comment (slabserif) [inline src/Core/compiler/resources/pokedex.txt]: supposedly actual base stats. Doesn't seem right, need to research more.

**Pull request merged**